packages/spotlight/src/server/sdk.ts

```diff
@@ -0,0 +1,62 @@
+import { brotliDecompressSync, gunzipSync, inflateSync } from "node:zlib";
+import { MessageBuffer } from "./messageBuffer.ts";
+import { EventContainer } from "./utils/eventContainer.ts";
+
+const decompressors: Record<string, ((buf: Buffer) => Buffer) | undefined> = {
+  gzip: gunzipSync,
+  deflate: inflateSync,
+  br: brotliDecompressSync,
+};
+
+export function createSpotlightBuffer() {
+  return new MessageBuffer<EventContainer>();
+}
+
+type PushToSpotlightBufferOptions = {
+  spotlightBuffer: MessageBuffer<EventContainer>;
+  body: Buffer;
+  overrideBufferDecoding?: boolean;
+  headers: {
+    contentEncoding?: string;
+    contentType?: string;
+    userAgent?: string;
+    origin?: string;
+  };
+  query: {
+    sentry_client?: string;
+  };
+};
+
+export function pushToSpotlightBuffer(options: PushToSpotlightBufferOptions) {
+  const body = options.overrideBufferDecoding
+    ? options.body
+    : decompressBody(options.body, options.headers.contentEncoding);
+
+  let contentType = options.headers.contentType?.split(";")[0].toLocaleLowerCase();
+  if (options.query.sentry_client?.startsWith("sentry.javascript.browser") && options.headers.origin) {
+    // This is a correction we make as Sentry Browser SDK may send messages with text/plain to avoid CORS issues
+    contentType = "application/x-sentry-envelope";
```

Comment:
We should move this check outside as it is specific to the server implementation. It has no place in the SDK API.